### PR TITLE
131 make a simulated detector that can write hdf files

### PR DESCRIPTION
**Pull request by stan-dot**

### CODE REVIEW

File: src/ophyd_async/sim/SimPatternDetector.py
Comment (coretl) [inline src/ophyd_async/sim/SimPatternDetector.py:1]: ditto

File: src/ophyd_async/sim/SimPatternDetector.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from ophyd_async.core import DirectoryProvider, StaticDirectoryProvider
+from ophyd_async.core.detector import StandardDetector
+from ophyd_async.core.signal import SignalR
+from ophyd_async.sim.PatternGenerator import PatternGenerator
+
+from .SimPatternDetectorControl import SimPatternDetectorControl
+from .SimPatternDetectorWriter import SimPatternDetectorWriter
+
+
+class SimDetector(StandardDetector):
```
Comment (coretl) [inline src/ophyd_async/sim/SimPatternDetector.py:13]:
```suggestion
class SimPatternDetector(StandardDetector):
```

**Review by coretl — CHANGES_REQUESTED** [review]
Couple of naming changes, other than that looks good

File: src/ophyd_async/sim/PatternGenerator.py
[the remark this answers is not in the document]
Comment (stan-dot) [reply src/ophyd_async/sim/PatternGenerator.py:1]: ah of course

[the remark this answers is not in the document]
Comment (stan-dot) [reply src/ophyd_async/sim/PatternGenerator.py:1]: why not snake case? given the other files?
`src/ophyd_async/epics/
├── __init__.py
├── _backend
│   ├── __init__.py
│   ├── _aioca.py
│   └── _p4p.py
├── areadetector
│   ├── __init__.py
│   ├── controllers
│   │   ├── __init__.py
│   │   ├── ad_sim_controller.py
│   │   └── pilatus_controller.py
│   ├── drivers
│   │   ├── __init__.py
│   │   ├── ad_base.py
│   │   └── pilatus_driver.py
│   ├── single_trigger_det.py
│   ├── utils.py
│   └── writers
│       ├── __init__.py
│       ├── _hdfdataset.py
│       ├── _hdffile.py
│       ├── hdf_writer.py
│       ├── nd_file_hdf.py
│       └── nd_plugin.py
├── demo
│   ├── __init__.py
│   ├── demo_ad_sim_detector.py
│   ├── mover.db
│   └── sensor.db
├── motion
│   ├── __init__.py
│   └── motor.py
└── signal
    ├── __init__.py
    ├── _epics_transport.py
    ├── pvi_get.py
    └── signal.py

9 directories, 29 files`

File: src/ophyd_async/sim/SimPatternDetector.py (hunk 1 above)
Comment (stan-dot) [reply src/ophyd_async/sim/SimPatternDetector.py:13]: done

File: src/ophyd_async/sim/PatternGenerator.py
[the remark this answers is not in the document]
Comment (coretl) [reply src/ophyd_async/sim/PatternGenerator.py:1]: PEP8 says snake case is also fine for modules (although not for packages), so `pattern_generator.py` is fine

[the remark this answers is not in the document]
Comment (stan-dot) [reply src/ophyd_async/sim/PatternGenerator.py:1]: ok will do after the meeting

**Review by coretl — APPROVED** [review]

**Comment by stan-dot** [comment]
that issue is still present with pyside6 at 6.7.0
https://bugreports.qt.io/browse/PYSIDE-2665?jql=text%20~%20"unexpected%20indent%22

**Pull request merged**